engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -1235,4 +1235,30 @@ function updateLastAction($guid, $posted = null) {
 			return false;
 		}
 	}
+
+	/**
+	 * Get a user by GUID even if the entity is hidden or disabled
+	 *
+	 * @param int $guid User GUID. Default is logged in user
+	 *
+	 * @return \ElggUser|false
+	 * @access private
+	 */
+	public function resolveUser($guid = 0) {
+		if (!$guid) {
+			return _elgg_services()->session->getLoggedInUser();
+		}
+
+		// need to ignore access and show hidden entities for potential hidden/disabled users
+		$ia = _elgg_services()->session->setIgnoreAccess(true);
+		$show_hidden = access_show_hidden_entities(true);
+
+		$user = $this->get($guid, 'user');
```

Comment:
Type is enforced in https://github.com/Elgg/Elgg/blob/2.x/engine/classes/Elgg/Database/EntityTable.php#L140